dotnet/src/InternalUtilities/src/Diagnostics/ActivityExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel.Diagnostics;
+
+[ExcludeFromCodeCoverage]
+internal static class ActivityExtensions
+{
+    public static Activity? StartActivityWithTags(this ActivitySource source, string name, List<KeyValuePair<string, object?>> tags)
+    {
+        return source.StartActivity(
+            name,
+            ActivityKind.Internal,
```

Comment:
according to the semconv it should be client. I'm violating it in my prototypes, not sure what to do about it for now, but you don't have to violate :)